### PR TITLE
feat(minifier): fold string string comparison

### PR DESCRIPTION
**Pull request by mysteryven**

### CODE REVIEW

File: crates/oxc_hir/src/hir_util.rs [hunk 1]
```diff
@@ -372,3 +376,66 @@ pub fn get_boolean_value(expr: &Expression) -> Option<bool> {
         _ => None,
     }
 }
+
+/// Port from [closure-compiler](https://github.com/google/closure-compiler/blob/e13f5cd0a5d3d35f2db1e6c03fdf67ef02946009/src/com/google/javascript/jscomp/NodeUtil.java#L234)
+/// Gets the value of a node as a String, or `None` if it cannot be converted. When it returns a
+/// String, this method effectively emulates the `String()` JavaScript cast function.
+/// This method does not consider whether `expr` may have side effects.
+fn get_string_value(expr: &Expression) -> Option<String> {
```
Comment (Boshen) [inline crates/oxc_hir/src/hir_util.rs:384]: It is better to return `&str` instead of an allocated `String`

Comment (Boshen) [inline crates/oxc_hir/src/hir_util.rs:384]: It is better to return &str instead of an allocated String

Comment (Boshen) [reply crates/oxc_hir/src/hir_util.rs:384]: It seems you need to return allocated string in some cases? Try `get_string_value<'a>(expr: &'a Expression) -> Option<Cow<'a, str>>`

**Review by Boshen — CHANGES_REQUESTED** [review]

File: crates/oxc_hir/src/hir_util.rs [hunk 2]
```diff
@@ -371,3 +377,65 @@ pub fn get_boolean_value(expr: &Expression) -> Option<bool> {
...
+                        .map(|boolean| (!boolean).to_string().into())
+                }
+                _ => None,
+            }
+        }
+        Expression::ArrayExpression(_) => {
+            // TODO: https://github.com/google/closure-compiler/blob/e13f5cd0a5d3d35f2db1e6c03fdf67ef02946009/src/com/google/javascript/jscomp/NodeUtil.java#L302-L303
+            None
+        }
+        Expression::ObjectExpression(_) => Some("[object Object]".to_string().into()),
```
Comment (Boshen) [inline crates/oxc_hir/src/hir_util.rs:422]: Does Cow::owned or Cow::borrowed work? This is static so it has a longer lifetime

**Review by Boshen — APPROVED** [review]

**Comment by mysteryven** [comment]
I have refactored it. Could you please take a look?

**Pull request merged**